### PR TITLE
test peer destroy()

### PR DESCRIPTION
**Pull request by volstas**
First run shows how existing behavior fails:
https://github.com/holepunchto/blind-peer/actions/runs/31001800366/job/92292186377

AI usage:
This is a bug found by AI. Confirmed the behavior. Half of the reported issue was expected, but half is real. 
The test is simplified a lot from what AI generated, and additionally tests the accumulation of listeners over same core when that core is added multiple times(the core itself is deduped, but listeners accumulate in some specific conditions).

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -3045,6 +3045,60 @@ test('sendNotification does not create a second ref to an already-added blind pe
...
+  t.is(base.core.listenerCount('migrate'), 0, 'base core 0 "migrate" liteners initially')
+
+  client.addCoreBackground(core, { pick: 5 })
+  client.addCoreBackground(core, { pick: 10 })
+  await client.addAutobase(base)
+
+  const peer = client.blindPeers.get(b4a.toString(blindPeer.publicKey, 'hex'))
+
+  t.is(peer.cores.size, 1, '1 core is added despite adding it twice')
+  t.is(peer.cores.values().next().value.pick, 10, 'info object is from the second addition')
```
Comment (HDegroote) [inline test/test.js]: <strike>Is this relevant to the bug? ATM we have no strictly defined behaviour on what happens if you add a core a second time with different opts, so would not include that assertion unless needed</strike>

Makes sense now that I reviewed the blind-peering PR more in depth, since that's a symptom of the explicit edge case where we leak the sessions

**Review by HDegroote — COMMENTED** [review]
Test looks clean and minimal, just the one clarification

File: test/test.js [hunk 2]
```diff
@@ -3045,6 +3045,74 @@ test('sendNotification does not create a second ref to an already-added blind pe
...
+  t.is(base.core.listenerCount('migrate'), 1, 'base core 1 "migrate" listener after adding')
+
+  const peer = client.blindPeers.get(b4a.toString(blindPeer.publicKey, 'hex'))
+
+  await client.close()
+
+  t.is(peer.destroyed, true, 'closing blind-peering destroyed the peer')
+  t.is(base.listenerCount('close'), 0, 'base 0 "close" listeners after peer is destroyed')
+  t.is(
+    base.core.listenerCount('migrate'),
```
Comment (HDegroote) [inline test/test.js:3112]: Can you also sanity check the 'writer' listener is at 0? Ten that's tracked too

**Review by HDegroote — APPROVED** [review]
